src/shared/Core/ChildProcess.cs

```diff
@@ -19,26 +20,38 @@ public class ChildProcess : DisposableObject
     public StreamWriter StandardInput => Process.StandardInput;
     public StreamReader StandardOutput => Process.StandardOutput;
     public StreamReader StandardError => Process.StandardError;
-    public int Id => Process.Id;
     public int ExitCode => Process.ExitCode;
 
-    public static ChildProcess Start(ITrace2 trace2, ProcessStartInfo startInfo)
+    public static ChildProcess Start(ITrace2 trace2, ProcessStartInfo startInfo, Trace2ProcessClass processClass)
     {
         var childProc = new ChildProcess(trace2, startInfo);
-        childProc.Start();
+        childProc.Start(processClass);
         return childProc;
     }
 
     public ChildProcess(ITrace2 trace2, ProcessStartInfo startInfo)
     {
         _trace2 = trace2;
         Process = new Process() { StartInfo = startInfo };
+        Process.Exited += ProcessOnExited;
     }
 
-    public void Start()
+    public void Start(Trace2ProcessClass processClass)
     {
         ThrowIfDisposed();
+        // Record the time just before the process starts, since:
+        // (1) There is no event related to Start as there is with Exit.
+        // (2) Using Process.StartTime causes a race condition that leads
+        // to an exception if the process finishes executing before the
+        // variable is passed to Trace2.
+        _startTime = DateTimeOffset.UtcNow;
```

Comment:
You could also say that we want to include the process startup cost in the timing.  
In Git, I write the child-start message before I spawn the child and the child-exit message after returning from wait3() or WaitForSingleObject(), so that we get the outer timing and all of the process overhead.

Your way is fine -- your child-start message will execute concurrently with the child startup, so it'll be slightly more efficient.  The important thing is using the clock value from before you called Process.Start().

Reply:
Going to leave as is for now to lessen the (admittedly already significant) churn in this PR.